logic/resolvable/Concludable.java

```diff
@@ -101,25 +97,9 @@ public Concludable asConcludable() {
 
     public abstract Set<Constraint> concludableConstraints();
 
-    public FunctionalIterator<Unifier> getUnifiers(Rule rule) {
-        assert applicableRules != null;
-        return iterate(applicableRules.get(rule));
-    }
-
-    public FunctionalIterator<Rule> getApplicableRules(ConceptManager conceptMgr, LogicManager logicMgr) {
-        if (applicableRules == null) {
-            synchronized (this) {
-                if (applicableRules == null) applicableRules = applicableRules(conceptMgr, logicMgr);
-            }
-        }
-        // This gives a deterministic ordering to the applicable rules, which is important for testing.
-        return Iterators.iterate(applicableRules.keySet().stream().sorted(Comparator.comparing(Rule::getLabel))
-                                         .collect(Collectors.toList()));
-    }
-
-    abstract Map<Rule, Set<Unifier>> applicableRules(ConceptManager conceptMgr, LogicManager logicMgr);
+    public abstract Map<Rule, Set<Unifier>> applicableRules(ConceptManager conceptMgr, LogicManager logicMgr);
```

Comment:
Since the applicable rules are determined by unification and the type of conclusion (Isa, Has, Relation etc), the Concludable still has to determine this. It is called by the new LogicManager.applicableRules

Reply:
`computeApplicableRules` to indicate it's a potentially heavy operation